### ISSUE TITLE
Define data structure and form questions

### ISSUE THREAD

**Issue by tuttiq** [WWCodeTokyo/speak-her-db#13]
- [x] Define/update [table columns on Airtable](https://airtable.com/tbliScQ1NqsY6pPcf/viwqShTXXyw6kwG6j)
- [x] Update [Form questions](https://airtable.com/tbliScQ1NqsY6pPcf/viwHeyq6Vo8jRIZqY) accordingly (migrate from [Google Form](https://docs.google.com/forms/d/1NQu2yG_ril6KfSMCSKEYFhgz3wcEpFV09NbFMtdQR9E/edit?ts=5eb53e7d))

**Comment by tuttiq** [comment]
@ann-kilzer @yanarchy 

Here's the [latest version of our database](https://airtable.com/tbliScQ1NqsY6pPcf/viwqShTXXyw6kwG6j?blocks=hide).

### DB specs

- Tables: `People`, `Location` and `Topics`
- Added all Japanese prefectures on the table `Location`.
- Added a few sample topics on table `Topics`, with a respective `field` (like "STEM" or "Arts")
- Views for the `People` table:
    - `Published` - only records with `published = true`
        - This flag will allow us to review submissions before they show on the website. Once we review them, we can manually toggle `published` to make it `true`
        - On this view, the field `email` is hidden
        - We should use this view to query the list of speakers to show on the website
    - `By Topic` - just a view of speakers grouped by topic
    - `By Location` - just a view of speakers grouped by prefecture
    - `Submit A Speaker` - our [submission form](https://airtable.com/shrOSFTp1mDij00AK)

### Known issues

- The `Topics` dropdown on the submission form only show topic names, without the corresponding `Field`. This becomes a problem when we have Topics with same/similar names in more than one Field (e.g. `Leadership & Management` under `STEM` and under `Business`)
- We may need to rethink this whole approach to topics, because apparently we can't allow people to add topics on the fly while filling out the form...

**Alternative ideas**
- Let `topics` be a free paragraph where the speaker can input however they want to illustrate the topics they can speak about (paragraph, comma separated values, whatever)
- Add a `keywords` field that the speaker can add keywords separated by comma to help people find their profile
- On the site search, the search query should look at both these fields (`topics` and `keywords`) to find the search results

----------------

What do you think? Any other ideas?

**Comment by ann-kilzer** [comment]
Looking great. Thanks for organizing this! 

- I like the choice to use the "published" column. Can we restrict API users to only use this view, or is this going to be where our backend proxy does the protecting?
- Would suggest adding a column for city, as some prefectures are quite large.
- Previous companies instead of additional_company?
- On topics:
  - My 2 cents is the field with topics is too hierarchical. What about just making one tier, and consider them as an "AND" function. 
  - I think topics vs. keywords could confuse the user. Is there a functional difference between the two? Aren't both ways to narrow down candidates based on a word or two?
  - Could we have a separate input for submitting new topics?

**Comment by tuttiq** [comment]
> I like the choice to use the "published" column. Can we restrict API users to only use this view, or is this going to be where our backend proxy does the protecting?

Airtable doesn't have access control per view. But to access the database via API, anyone would need our database ID and have an authorized API Key (only accounts we add on our DB settings have authorized API Keys, so as of now, only the 3 of us).

We are currently storing this info on the env variables and we should be careful to never publish these anywhere.

But like we talked last time, for an extra layer of security, we can have:
- Move these env variables to the backend
- Only access Airtable using the `Published` view, that has no contact info
- Added a security hash to the main view name (the one that has contact info), so to access it via API you also have to know this hash

**Comment by tuttiq** [comment]
> Would suggest adding a column for city, as some prefectures are quite large.

Ok, we can do that.

> Previous companies instead of additional_company?

In the form this field is "Former or additional company/organization". The reason is for cases like us, where we have a main job at a main company, but are also part of an additional relevant organization, like Women Who Code.

I guess some people may choose to refer their former company if they think it's relevant too (e.g. "Ex-Google"), but I was assuming most cases would be an additional organization (non-profits, user groups, communities, associations, etc). 

What do you think?

**Comment by tuttiq** [comment]
> On topics:
> - My 2 cents is the field with topics is too hierarchical. What about just making one tier, and consider them as an "AND" function.
> - I think topics vs. keywords could confuse the user. Is there a functional difference between the two? Aren't both ways to narrow down candidates based on a word or two?

Agree. I changed it to just `topics` and it's a normal string value, with [instructions on the form](https://airtable.com/shrOSFTp1mDij00AK) to enter topics separated by comma (so we can programmatically split it and treat them as tags/keywords)

Additionally, I added a dropdown to select `industry` (from a populated list). In my opinion it's a "nice to have", but not extremely necessary, so let me know what you think, if this field should stay or go :)

> Could we have a separate input for submitting new topics?

Unfortunately not, this is one of the limitations of using Airtable for this MVP apparently. It's either a free text field or a limited set of pre-created options 😞

Let me know if you think the spec I described above (free string, values separated by comma) is an acceptable spec for the MVP.
